bin/client/src/kona.rs

```diff
@@ -58,7 +58,8 @@ fn main() -> Result<()> {
             l2_provider.clone(),
         )
         .await?;
-        let L2AttributesWithParent { attributes, .. } = driver.produce_disputed_payload().await?;
+        let L2AttributesWithParent { attributes, .. } =
+            driver.produce_disputed_payload().await?.expect("Failed to derive payload attributes");
```

Comment:
If we keep the signature `fn() -> Result<L2AttributesWithParent>` as I suggest, we shouldn't need to change this since the error will bubble up :)